stager_test.go

```diff
@@ -262,13 +262,37 @@ var _ = Describe("Stager", func() {
 
 	Describe("AddBinDependencyLink", func() {
 		It("creates a symlink <depDir>/bin/<name> with the relative path to dest", func() {
-			err := s.AddBinDependencyLink(filepath.Join(depsDir, depsIdx, "some", "long", "path"), "dep")
-			Expect(err).To(BeNil())
+			var err error
+			destDir := filepath.Join(depsDir, depsIdx, "some", "long")
+			dest := filepath.Join(destDir, "path")
+
+			/* Windows uses hard links, the file must already exist and
+			* be an actual file (not a directory)
+			 */
+			if runtime.GOOS == "windows" {
+				err = os.MkdirAll(destDir, 0777)
+				Expect(err).To(BeNil())
+				f, err := os.Create(dest)
+				Expect(err).To(BeNil())
+				f.Close()
+			}
 
-			link, err := os.Readlink(filepath.Join(s.DepDir(), "bin", "dep"))
+			err = s.AddBinDependencyLink(dest, "dep")
 			Expect(err).To(BeNil())
 
-			Expect(link).To(Equal("../some/long/path"))
+			linkSource := filepath.Join(s.DepDir(), "bin", "dep")
+
+			if runtime.GOOS == "windows" {
+				f1, err := os.Stat(linkSource)
+				Expect(err).To(BeNil())
+				f2, err := os.Stat(dest)
+				Expect(err).To(BeNil())
+				Expect(os.SameFile(f1, f2)).To(BeTrue())
+			} else {
+				link, err := os.Readlink(linkSource)
+				Expect(link).To(Equal(filepath.Join("..", "some", "long", "path")))
+				Expect(err).To(BeNil())
```

Comment:
Checking this error is nil should come before checking the value of `link` because when a function returns an error, the rest of its output is generally invalid.